charon/src/ast/types_utils.rs

```diff
@@ -630,6 +630,73 @@ impl Ty {
         }
     }
 
+    pub fn get_ptr_metadata(&self, translated: &TranslatedCrate) -> PtrMetadata {
+        let ref ty_decls = translated.type_decls;
+        let ref trait_decls = translated.trait_decls;
+        match self.kind() {
+            TyKind::Adt(ty_ref) => {
+                // there are two cases:
+                // 1. if the declared type has a fixed metadata, just returns it
+                // 2. if it depends on some other types or the generic itself
+                match ty_ref.id {
+                    TypeId::Adt(type_decl_id) => {
+                        let Some(decl) = ty_decls.get(type_decl_id) else {
+                            panic!(
+                                "Internal Error: type decl id not found during getting metadata: {type_decl_id}"
+                            )
+                        };
```

Comment:
Please use `raise_error`, this can happen in normal usage of charon such as with `--opaque`. `panic` should be only used for cases that are theoretically impossible.